Add nullability annotations to public API

fractalwrench opened the pull request:
Adds nullability annotations to the public API in the `leakcanary-android` module, which improves Kotlin support. This changeset alters:

- Android Gradle Plugin bumped to `3.2.0-rc03` (the `UnknownNullness` lint inspection was only introduced in `3.2.0-alpha10`)
- Enables [Kotlin interoperability checks](https://android.github.io/kotlin-guides/interop.html#command-line-builds) for the `leakcanary-android` module

Points of discussion:

- I've avoided adding annotations to the other android library modules which have public classes, as that would require adding the support library, and I'm unfamiliar with the project's architecture. If that's not an issue, I can update the PR to annotate those classes also
- I've avoided making `UnknownNullness` fail the build, as the plan was only to add annotations to the public API surface
- The Gradle plugin version might need to be bumped to `v3.2.0` once that's released

To test this change I auto-converted the sample project to Kotlin and ensured that passing null to public methods on the `LeakCanary` class resulted in compiler methods, on a debug build. I also ran `./gradlew clean build` which passed locally.

Give me a shout if you have any questions or comments and I'll be happy to respond.

Fixes #1016

JakeWharton commented on leakcanary-android/src/main/java/com/squareup/leakcanary/DefaultLeakDirectoryProvider.java:
```diff
@@ -19,6 +19,9 @@
 import android.os.Environment;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
```
The code styles will also eliminate all these newlines in imports

JakeWharton commented on leakcanary-watcher/src/main/java/com/squareup/leakcanary/HeapDumper.java:
```diff
@@ -15,6 +15,8 @@
  */
 package com.squareup.leakcanary;
 
+import com.sun.istack.internal.Nullable;
```
Wrong import

jrodbx commented:
> as that would require adding the support library
Adding support-annotations is fine.  It's only a handful of java annotations.

jrodbx commented on leakcanary-android/src/main/java/com/squareup/leakcanary/AndroidExcludedRefs.java:
```diff
@@ -590,7 +592,7 @@
    * This returns the references in the leak path that should be ignored by all on Android.
    */
-  public static ExcludedRefs.Builder createAndroidDefaults() {
+  @NonNull public static ExcludedRefs.Builder createAndroidDefaults() {
```
nit: for return values, apply @NonNull and @Nullable just before the type, i.e., `public static @NonNull ExcludedRefs.Builder`.

fractalwrench commented:
Thanks for all the feedback so far! I have made a few more changes to this PR, namely:

- Code style cleanup by installing SquareAndroid as suggested
- Fixed imports
- Add nullability annotation directly before type for methods which return non-void values
- Added nullability annotations to public API in all other Android library modules
